GatorEducator/gatorgrade
# Code update: refactor the way the report files are defined in the command line and passed into the configure_reports function

# Code update: refactor the way the report files are defined in the command line and passed into the configure_reports function

## Description
This latest change to `gatorgrade` is to refactor the code that configures the report. This update changes the command line interface and the commands that are passed in. There is no longer a `--report`` command rather there are three that address different aspects.
The purpose of this is to have better documentation for users to understand the acceptable options for each part of generating a report.

## Linked Issues
[https://github.com/GatorEducator/gatorgrade/issues/145](url)

closes: #145 

## Type of Change

## Contributors
- @rebekahrudd 

## Testing this PR 

Setup:
1. Choose any unfinished repo with `gatorgrade` to test this `gatorgrade` update.
2. In the `pyproject.toml` file of the repo you chose add the command `pipx install git+https://github.com/rebekahrudd/gatorgrade` in place of the command `pipx install gatorgrade`
3. Save and push this update to Github
4. Wait for Github actions to run

How to check if this PR worked:
1. In GitHub actions check Gatorgrade to ensure you see "This is the new update of gatorgrade" printed next to FAILURES. This way you will know that rebekahrudd's branch ran and not the latest pipx version of `gatorgrade`
2. When you are on the summary page of Github actions you should be able to see the GatorInsights Report. It should look like this:
![Screenshot from 2024-10-24 19-30-41](https://github.com/user-attachments/assets/7b127cd5-8798-48f7-a956-462d043a18b7)

If you see the report and have confirmed that it was my version of Github that ran then this update is working and this PR can get approved!
